dwave/cloud/utils.py

```diff
@@ -801,3 +802,197 @@ def bqm_to_dqm(bqm):
+
+    if anneal_schedule and anneal_schedule[0][1] == 1 and not initial_state:
+        raise ValueError("reverse annealing requested (``anneal_schedule`` "
+                         "starts with s==1) but ``initial_state`` not provided")
+
+    try:
+        problem_timing_data = solver.properties['problem_timing_data']
+
+        version_timing_model = problem_timing_data['version']
+        typical_programming_time = problem_timing_data['typical_programming_time']
+        ra_with_reinit_prog_time_delta = problem_timing_data['reverse_annealing_with_reinit_prog_time_delta']
+        ra_without_reinit_prog_time_delta = problem_timing_data['reverse_annealing_without_reinit_prog_time_delta']
+        default_programming_thermalization = problem_timing_data['default_programming_thermalization']
+        default_annealing_time = problem_timing_data['default_annealing_time']
+        readout_time_model = problem_timing_data['readout_time_model']
+        readout_time_model_parameters = problem_timing_data['readout_time_model_parameters']
+        qpu_delay_time_per_sample = problem_timing_data['qpu_delay_time_per_sample']
+        ra_with_reinit_delay_time_delta = problem_timing_data['reverse_annealing_with_reinit_delay_time_delta']
+        ra_without_reinit_delay_time_delta = problem_timing_data['reverse_annealing_without_reinit_delay_time_delta']
+        decorrelation_max_nominal_anneal_time = problem_timing_data['decorrelation_max_nominal_anneal_time']
+        decorrelation_time_range = problem_timing_data['decorrelation_time_range']
+        default_readout_thermalization = problem_timing_data['default_readout_thermalization']
+
+    except KeyError as err:
+        if err == 'problem_timing_data':
+            err_msg = f"selected solver is missing required property {err}"
+        elif err == 'version':
+            err_msg = f"selected solver is missing ``problem_timing_data`` field {err}"
+        else:
+            err_msg = f"selected solver is missing ``problem_timing_data`` field " + \
+                      f"{err} required by timing model version {version_timing_model}"
+        raise ValueError(err_msg)
+
+    # Support for sapi timing model versions: 1.0.x
+    version_tuple = tuple(int(i) for i in version_timing_model.split("."))
+    if not version_tuple[0] == 1 and not version_tuple[1] == 0:
+        raise ValueError(f"``estimate_qpu_access_time`` does not currently "
+                         f"support timing model {version_timing_model} "
+                         "used by the selected solver")
+
+    ra_programming_time = 0
+    ra_delay_time = 0
+    if anneal_schedule and anneal_schedule[0][1] == 1:
+        if reinitialize_state:
+            ra_programming_time = ra_with_reinit_prog_time_delta
+            ra_delay_time = ra_with_reinit_delay_time_delta
+        else:
+            ra_programming_time = ra_without_reinit_prog_time_delta
+            ra_delay_time = ra_without_reinit_delay_time_delta
```

Comment:
Replace `=` with `+=` for clarity and future-proofing (moving code sections around).

Reply:
Can you clarify? These are not accumulating, they're being set to the property value versus (currently) being zeroed; if in the future the potential for RA always adds a small constant bias, replacing = with += will instead introduce an error.

Reply:
Ok, in that case, it's fine like this. It's just when I see 2 variables initialized in 3 places, I want to refactor -- to initialize only once, and then update as needed. Same effect here, though.